system/handlers/Login.cfc

```diff
@@ -107,7 +107,7 @@ component {
 		}
 
 		args.allowRememberMe = _getRememberMeAllowed();
-		args.postLoginUrl    = websiteLoginService.getPostLoginUrl( rc.postLoginUrl ?: event.getCurrentUrl() );
+		args.postLoginUrl    = !isEmpty( rc.postLoginUrl ?: "" ) ? rc.postLoginUrl : websiteLoginService.getPostLoginUrl( rc.postLoginUrl ?: event.getCurrentUrl() );
```

Comment:
I think this needs a little more work. Looking at the `getPostLoginUrl()` method shows that it sets this value into your session scope if its not already there and this will stop happening with this change. I suggest:

```
args.postLoginUrl = websiteLoginService.getPostLoginUrl(
      explicitUrl = rc.postLoginUrl ?: ""
    , defaultUrl  = event.getCurrentUrl()
);
```

Then change the logic in `websiteLoginService.getPostLoginUrl()` to always use the `explicitUrl` if it is not empty + to set it into session if using it. Otherwise, try using what is in session. Otherwise, use the defaultUrl and set it into the session.

Reply:
Hi @DominicWatson  I have made a changes accordingly. It will now pickup explicitURL as priority, save it if available, otherwise look for the value from session or default value